lib/src/mt_ptp.c

```diff
@@ -51,6 +57,15 @@ static inline int ptp_get_time_spec(struct mt_ptp_impl* ptp, struct timespec* sp
   return ret;
 }
 
+static inline uint64_t ptp_get_correct_time(struct mt_ptp_impl* ptp) {
+  struct timespec spec;
+
+  ptp_get_time_spec(ptp, &spec);
+  uint64_t ts = mt_timespec_to_ns(&spec);
+  if (ptp->coefficient != 1.0) ts = ptp_correct_ts(ptp, ts);
```

Comment:
Is it "ptp->coefficient != 1.0" working for a double?

Reply:
nope, double should be compared using math library, removed this